src/clappr-dash-shaka-playback.js

```diff
@@ -140,19 +150,42 @@ class DashShakaPlayback extends HTML5Video {
     return (this.isReady && this._player.isLive() ? 'live' : 'vod') || ''
   }
 
-  selectTrack (track) {
-    if (track.type === 'text') {
-        this._player.selectTextTrack(track)
-    } else if (track.type === 'variant') {
-        this._player.selectVariantTrack(track)
-        if (track.mimeType.startsWith('video/')) {
-            // we trigger the adaptation event here
-            // because Shaka doesn't trigger its event on "manual" selection.
-            this._onAdaptation()
-        }
-    } else {
-        throw new Error('Unhandled track type:', track.type);
+  selectTrack (track, clearBuffer) {
```

Comment:
I don't see too much value by changing an `if else if else`  by a switch, there was another reason for this refactoring?

Reply:
hm well no there is no change in logic.

from a code-style pov i would say that a switch is preferrable here since the condition is solely around the value of a single variable and no other conditions, and there are more than two cases :) so perfect time for a "switch"

so my answer is: this shouldn't have been an if-else in the first place. although i had written this code right there, just correcting myself now ;)

Reply:
Yes, I see your points :) Later I'll try to break this single class into at least 4 different components.

Reply:
i think it's still quite maintainable like this.

we should however add tests + enforce JSdocs with the eslint plugin for that on each method :)

Reply:
Yeah, tests are way up on the priority list. 🥂